plugins/pdr_deterministic_plugin/ufm_sim_web_service/isolation_mgr.py

```diff
@@ -231,171 +286,121 @@ def read_next_set_of_high_ber_or_pdr_ports(self, endpoint_port):
-                        port_data[Constants.ASIC] = port_asic
-                        port_data[Constants.WIDTH] = port_width
-                    if not port_asic:
-                        self.logger.debug(f"Couldn't retrieve HW technology for port {port_name}, can't verify it's BER values")
-                        continue
-                    if not port_width:
+                    port_obj.ber_tele_data.loc[len(port_obj.ber_tele_data)] = ber_data
+                    port_obj.last_symbol_ber_timestamp = timestamp
+                    port_obj.last_symbol_ber_val = symbol_ber_val
+                    if not port_obj.active_speed or not port_obj.port_width:
+                        port_obj.active_speed, port_obj.port_width = self.get_port_metadata(port_name)
+                    if not port_obj.port_width:
                         self.logger.debug(f"port width for port {port_name} is None, can't verify it's BER values")
                         continue
-
-                    raw_ber_rate, eff_ber_rate, symbol_ber_rate = self.calc_ber_rates(port_name, port_speed, port_width)
-                    if (raw_ber_rate or eff_ber_rate or symbol_ber_rate) and self.check_ber_threshold(port_name, port_speed, port_asic, fec_mode, raw_ber_rate, eff_ber_rate, symbol_ber_rate, isolation=True):
-                        issued_port = issues.get(port_name)
-                        if issued_port:
-                            issued_port.cause = Constants.ISSUE_PDR_BER
-                        else:
-                            issues[port_name] = Issue(port_name, Constants.ISSUE_BER)                    
+                    for (interval, threshold) in Constants.BER_THRESHOLDS_INTERVALS:
+                        symbol_ber_rate = self.calc_ber_rates(port_name, port_obj.active_speed, port_obj.port_width, interval)
+                        if symbol_ber_rate and symbol_ber_rate > threshold:
+                            issued_port = issues.get(port_name)
+                            if issued_port:
+                                issued_port.cause = Constants.ISSUE_PDR_BER
+                            else:
+                                issues[port_name] = Issue(port_name, Constants.ISSUE_BER)                    
+                            break                        
         return issues
 
-    def calc_single_rate(self, port_name, port_speed, port_width, col_name):
+    def calc_single_rate(self, port_name, port_speed, port_width, col_name, time_delta):
         try:
-            min_timestamp = self.ber_tele_data.loc[self.ber_tele_data[Constants.PORT_NAME] == port_name, Constants.TIMESTAMP].min()
-            max_timestamp = self.ber_tele_data.loc[self.ber_tele_data[Constants.PORT_NAME] == port_name, Constants.TIMESTAMP].max()
-            self.ber_tele_data[Constants.TIMESTAMP].min()
-            if max_timestamp == min_timestamp or max_timestamp - min_timestamp < self.ber_wait_time:
+            if port_speed != "NDR":
```

Comment:
may be to do this check on stage before calling the function? more efficient

Reply:
it is called in two functions that test also other conditions related to other calculation that needs to be done so only for this symbol BER check we need to test this NDR